test/DynamoCoreWpfTests/DynamoTestUIBase.cs

```diff
@@ -20,10 +20,40 @@
 using DynamoShapeManager;
 using DynamoUtilities;
 using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
 using TestServices;
 
 namespace DynamoCoreWpfTests
 {
+    internal class IsolatedTestCommand(TestCommand innerCommand) : DelegatingTestCommand(innerCommand)
+    {
+        public override TestResult Execute(TestExecutionContext context)
+        {
+            var t = new Thread(() =>
+            {
+                context.CurrentResult = innerCommand.Execute(context);
+                Dispatcher.CurrentDispatcher?.InvokeShutdown();
+            });
+
+            t.SetApartmentState(ApartmentState.STA);
+
+            t.Start();
+            t.Join();
+            return context.CurrentResult;
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    internal class TestOnSeparateThreadAttribute : TestAttribute, IWrapSetUpTearDown
+    {
+        public TestCommand Wrap(TestCommand command)
+        {
+            return new IsolatedTestCommand(command);
+        }
+    }
+
```

Comment:
this is the main code chage

Reply:
Are those threads reused in this context?

Reply:
No. Each thread will be disposed after it is used

Reply:
Ugh, looks like there is cross thread access to the UI, probably some static UI resources or something similar…

Reply:
there are lots of static events and static singletons now. It's good for all of us to keep in mind that even if we don't intend to create multiple instances of a type in a shipped product we might need to do so during testing.